Scripts/Runtime/DelayedExecution/UpdateSources/AbstractUnityUpdateSource.cs

```diff
@@ -8,39 +8,63 @@ namespace Anvil.Unity.DelayedExecution
     /// </summary>
     /// <typeparam name="T">Unity requires a MonoBehaviour for it's Update functions, this generic is of type
     /// <see cref="AbstractUnityUpdateSourceMonoBehaviour"/></typeparam>
-    public abstract class AbstractUnityUpdateSource<T> : AbstractUpdateSource where T:AbstractUnityUpdateSourceMonoBehaviour
+    public abstract class AbstractUnityUpdateSource<T> : AbstractUpdateSource where T : AbstractUnityUpdateSourceMonoBehaviour
     {
         private GameObject m_UpdateSourceGameObject;
-        
+
         protected AbstractUnityUpdateSource()
         {
             m_UpdateSourceGameObject = new GameObject($"[UpdateSource - {typeof(T)}]");
             GameObject.DontDestroyOnLoad(m_UpdateSourceGameObject);
             m_UpdateSourceGameObject.transform.SetParent(UnityUpdateSourceGameObjectManager.UpdateSourceRoot);
+
             T updateSource = m_UpdateSourceGameObject.AddComponent<T>();
             updateSource.SetOnUpdateDispatchFunction(DispatchOnUpdateEvent);
+            updateSource.OnDisposing += UpdateSource_OnDisposing;
         }
 
         protected override void DisposeSelf()
         {
             if (m_UpdateSourceGameObject != null)
             {
+                m_UpdateSourceGameObject.GetComponent<T>().OnDisposing -= UpdateSource_OnDisposing;
                 GameObject.Destroy(m_UpdateSourceGameObject);
                 m_UpdateSourceGameObject = null;
             }
-            
+
             base.DisposeSelf();
         }
+
+        private void UpdateSource_OnDisposing()
+        {
+            // If the game object was disposed as part of Unity exiting play mode
+            // then having the MonoBehaviour version disposed is ok because the rest 
+            // of the application is being torn down anyway.
+#if UNITY_EDITOR
+            if (!UnityEditor.EditorApplication.isPlayingOrWillChangePlaymode)
+            {
+                // Calling dispose explicitly allows this instance to be removed from 
+                // UpdateHandleSourcesManager. This is important when DomainReload is disabled.
+                Dispose();
+                return;
+            }
+#endif
+
+            // This usually happens when m_UpdateSourceGameObject or one of its parents
+            // - Is accidentally destroyed by another process desroying GameObjects in the hierary.
```

Comment:
typo - hierarchy

Reply:
lol just suggest the change
```suggestion
            // - Is accidentally destroyed by another process destroying GameObjects in the hierarchy.
```